### DISCUSSION TITLE
Custom values in enumerations

### DISCUSSION THREAD

**Discussion by daniel-mezo** [webauthn4j/webauthn4j#619] [Q&A] [answered]
Hi @ynojima,

I've noticed that Enumations implemented in the library, always have a default section to create custom values. I wonder why that is? Why don't these enumerations restrict values to what the webauthn specification allows?

Example (see the default section):

```
    public static UserVerificationRequirement create(@NonNull String value) {
        AssertUtil.notNull(value, "value must not be null.");
        switch (value) {
            case "required":
                return REQUIRED;
            case "preferred":
                return PREFERRED;
            case "discouraged":
                return DISCOURAGED;
            default:
                return new UserVerificationRequirement(value);
        }
    }
```

I'd personally expect an `IllegalArgumentException` to be thrown, but I might be wrong and just curious.

**Comment by ynojima** [comment, marked as answer]
Since WebAuthn specification expects relying parties to be able to handle unknown values for future extensibility (see: 
https://www.w3.org/TR/webauthn-2/#sct-domstring-backwards-compatibility), 
 WebAuthn4J defines the types as classes to represent unknown values instead of using  Java enumeration types.

**Reply by daniel-mezo** [reply]
Thank you @ynojima, I was not aware of this section.
